src/ape_test/provider.py

```diff
@@ -66,8 +68,32 @@ def connect(self):
         )
         endpoints = {**API_ENDPOINTS}
         endpoints["eth"] = merge(endpoints["eth"], {"chainId": static_return(chain_id)})
-        tester = EthereumTesterProvider(ethereum_tester=self._evm_backend, api_endpoints=endpoints)
-        self._web3 = Web3(tester)
+        return EthereumTesterProvider(ethereum_tester=self._evm_backend, api_endpoints=endpoints)
+
+    @property
+    def auto_mine(self) -> bool:
+        return self.tester.ethereum_tester.auto_mine_transactions
+
+    def __setattr__(self, attr: str, value: Any) -> None:
+        # NOTE: Need to do this until https://github.com/pydantic/pydantic/pull/2625 is figured out
+        if attr == "auto_mine":
+            if value is True:
+                self.tester.ethereum_tester.enable_auto_mine_transactions()
+            elif value is False:
+                self.tester.ethereum_tester.disable_auto_mine_transactions()
+            else:
+                raise TypeError("Expecting bool-value for auto_mine setter.")
+        else:
+            super().__setattr__(attr, value)
```

Comment:
This should be solved in latest versions of pydantic: https://github.com/pydantic/pydantic/pull/5502

```suggestion
    @auto_mine.setter
    def set_auto_mine(self, value: Any) -> None:
        if value is True:
            self.tester.ethereum_tester.enable_auto_mine_transactions()
        elif value is False:
            self.tester.ethereum_tester.disable_auto_mine_transactions()
        else:
            raise TypeError("Expecting bool-value for auto_mine setter.")
```

Reply:
yes you're right, however i had to make it:

```
    @auto_mine.setter
    def auto_mine(self, value: Any) -> None:
```

else it was failing...
PyCharm gives me the warning:

```
Names of function and decorator don't match; property accessor is not created 
```
so that is how I knew to change the name.

but after doing so, it works!